Assets/Scripts/SlipTime/SlipTimeManager.cs

```diff
@@ -36,57 +36,111 @@ public class SlipTimeManager : MonoBehaviour
         /// <summary>
         /// The default amount of time it takes to charge one charge of SlipTime.
         /// </summary>
-        public float slipTimeChargeDuration = 3;
+        public float slipTimeCooldownDuration = 60f;
+
+        /// <summary>
+        /// How much SlipTime duration to subtract if the player moves during SlipTime.
+        /// </summary>
+        public float slipTimeMovementPenaltyCoefficient = 0.5f;
+
+        /// <summary>
+        /// The default amount of time to decrease the current cooldown of SlipTime by when bullet grazing.
+        /// </summary>
+        public float bulletGrazingCooldownDecreaseCoefficient = 0.5f;
+
+        /// <summary>
+        /// Counter to keep track of time spent in SlipTime.
+        /// </summary>
+        public float SlipTimeCounter { get; private set; }
+
+        /// <summary>
+        /// Counter to keep track of time spent charging SlipTime.
+        /// </summary>
+        public float ChargeTimeCounter { get; private set; }
+
+        /// <summary>
+        /// Whether the player is in SlipTime.
+        /// </summary>
+        public bool InSlipTime { get; private set; }
 
-        private bool inSlipTime;
-        private double slipTimeCounter;
-        private double chargeTimeCounter;
+        /// <summary>
+        /// Whether the player is charging a charge of SlipTime.
+        /// </summary>
+        public bool IsChargingSlipTime { get; private set; }
 
         private void Start()
         {
             slipTimeCharges = maxSlipTimeCharges;
+            InSlipTime = false;
+            IsChargingSlipTime = false;
+            SlipTimeCounter = slipTimeDuration;
+            ChargeTimeCounter = slipTimeCooldownDuration;
         }
 
-        // Update is called once per frame
+        // Update is called once per frame.
         private void Update()
         {
-            // Logic for entering SlipTime
-            if (Input.GetButtonDown("Fire2") && !inSlipTime && slipTimeCharges > 0)
+            // Logic for entering SlipTime.
+            if (Input.GetButtonDown("Fire2") && !InSlipTime && slipTimeCharges > 0)
             {
-                inSlipTime = true;
+                InSlipTime = true;
                 slipTimeCoefficient = slipTimeScalar;
                 slipTimeCharges -= 1;
             }
 
-            // Logic for exiting SlipTime at user prompt
-            else if (Input.GetButtonDown("Fire2") && inSlipTime)
+            // Logic for exiting SlipTime at user prompt.
+            else if (Input.GetButtonDown("Fire2") && InSlipTime)
             {
-                inSlipTime = false;
-                slipTimeCounter = 0f;
+                InSlipTime = false;
+                SlipTimeCounter = slipTimeDuration;
                 slipTimeCoefficient = 1f;
+                IsChargingSlipTime = true;
```

Comment:
Did we plan to have cooldown start after the first usage of sliptime? Is this intended? If so, can we discuss with @hirudan  if we want sliptime charging only to happen AFTER the first usage of sliptime? 

Personally feels a bit off to only begin cooldown when we first enter sliptime. As a player, I think its more awarding to see bullet grazing take into effect the first time I use sliptime.

Reply:
Charging the cooldown of a non-instant ability before it's duration is over is not standard for abilities (at least in the games I play), so that's why I did it this way. Happy to change it though if we want.

Reply:
I think let's try it this way first, and if we really don't like the feel of not being able to charge sliptime during sliptime, we can always go in and change it later. But I think in general, we should let the ability run its course before charging begins.

Reply:
Sounds good.